shared/constants/types/notifications.js

```diff
@@ -2,8 +2,16 @@
 import * as I from 'immutable'
 import type {Tab} from '../tabs'
 
-export type NotificationKeys = 'kbfsUploading'
-export type BadgeType = 'regular' | 'update' | 'badged' | 'uploading'
+export type NotificationKeys = 'kbfsUploading' | 'outOfSpace'
+export type BadgeType =
+  | 'regular'
+  | 'update'
+  | 'badged'
+  | 'error'
+  | 'uploading'
+  | 'badged-update'
+  | 'badged-uploading'
+  | 'badged-error'
```

Comment:
I'm a bit confused. Does this PR add a second icon to the widget, so there could be a badged count and a status icon at the same time? If so it seems like we could have get rid of badged state, and the `badged-` prefix, and rely on `desktopAppBadgeCount` to determine if we should show the number. Then `BadgeType` just controls the status icon. How does that sound?

Reply:
Yeah, it lets there be both at once. That seems reasonable, I think.